internal/pkg/validator/validator.go

```diff
@@ -16,112 +21,161 @@ import (
 	"github.com/keboola/keboola-as-code/internal/pkg/utils"
 )
 
-type Validation struct {
+const anonymousField = "__anonymous__"
+
+// Rule is custom validation rule/tag.
+type Rule struct {
 	Tag          string
 	Func         validator.Func
 	ErrorMessage string
 }
 
-func Validate(value interface{}, rules ...Validation) error {
-	return ValidateCtx(value, context.Background(), "dive", "", rules...)
+// Validate nested struct fields or slice items.
+func Validate(value interface{}, rules ...Rule) error {
+	return ValidateCtx(context.Background(), value, "dive", "", rules...)
+}
+
+// ValidateCtx validates any value.
+func ValidateCtx(ctx context.Context, value interface{}, tag string, namespace string, rules ...Rule) error {
+	v := newValidator()
+	v.registerRule(rules...)
+
+	// nolint: errorlint // library always returns validator.ValidationErrors
+	if err := v.validator.VarCtx(ctx, value, tag); err != nil {
+		return v.formatError(err.(validator.ValidationErrors), namespace, reflect.ValueOf(value))
+	}
+	return nil
+}
+
+type wrapper struct {
```

Comment:
Pridat som strukturu `wrapper`, tak je to o nieco prehladnejsie, a nemusim tolko vela parametrov odovzdavat medzi volaniami funkcii.